internal/guest/runtime/runc/container.go

```diff
@@ -389,6 +400,70 @@ func (c *container) startProcess(
 		}
 	}
 
+	// This is to enabling container logging in Bare Metal Containers. This is in experimental stage now and need to be revisited.
+	var stdoutFifoPipe, stderrFifoPipe *os.File
+	if annotations != nil {
+		pipeNameSuffix, exists := annotations["io.microsoft.bmc.logging.pipelocation"]
+		pipeDirectory := "/run/gcs/containerlogs/"
+		stdoutPipeName := pipeDirectory + pipeNameSuffix + "-stdout"
+		stderrPipeName := pipeDirectory + pipeNameSuffix + "-stderr"
+		os.MkdirAll(pipeDirectory, os.ModePerm)
+		if exists {
+			_, err = os.Stat(stdoutPipeName)
+			if err != nil {
+				// fifo pipe does not exist, create one
+				err = syscall.Mkfifo(stdoutPipeName, 0666)
+				if err != nil {
+					msg := "Error creating named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+					logrus.Infof(msg)
+					return nil, errors.Wrapf(err, msg)
+				}
+			}
+
+			_, err = os.Stat(stderrPipeName)
+			if err != nil {
+				// fifo pipe does not exist, create one
+				err = syscall.Mkfifo(stderrPipeName, 0666)
+				if err != nil {
+					msg := "Error creating named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+					logrus.Infof(msg)
+					return nil, errors.Wrapf(err, msg)
+				}
+			}
+
+			// pipe either exist before hand or we have created one above
+			stdoutFifoPipe, err = os.OpenFile(stdoutPipeName, os.O_RDWR|os.O_APPEND, os.ModeNamedPipe)
+			if err != nil {
+				msg := "Error opening named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+				logrus.Infof(msg)
+				return nil, errors.Wrapf(err, msg)
+			}
+
+			stderrFifoPipe, err = os.OpenFile(stderrPipeName, os.O_RDWR|os.O_APPEND, os.ModeNamedPipe)
+			if err != nil {
+				msg := "Error opening named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+				logrus.Infof(msg)
+				return nil, errors.Wrapf(err, msg)
+			}
+		}
+
+		isLoggingSideCarContainerStr, exists := annotations["io.microsoft.bmc.logging.isLoggingSideCarContainer"]
```

Comment:
I'm a little confused on exactly what this annotation is meant to indicate. It seems like it indicates both that a workload container is using a sidecar container to output logs to AND used to indicate which container is the sidecar container. Is that correct?

Reply:
this is to identify whether the container is workload container or logging container.. for workload container we will redirect the stdout/stderr and for logging side car we will pass the pipe FDs.